packages/block-library/src/image/index.php

```diff
@@ -217,20 +224,33 @@ function block_core_image_render_lightbox( $block_content, $block ) {
 			__( 'Enlarged image' )
 		)
 	);
-	$w->next_tag( 'img' );
-	$w->set_attribute( 'data-wp-init', 'effects.core.image.setCurrentSrc' );
-	$w->set_attribute( 'data-wp-on--load', 'actions.core.image.handleLoad' );
-	$w->set_attribute( 'data-wp-effect', 'effects.core.image.setButtonStyles' );
-	$w->set_attribute( 'data-wp-effect--setStylesOnResize', 'effects.core.image.setStylesOnResize' );
-	$body_content = $w->get_updated_html();
 
-	// Wrap the image in the body content with a button.
-	$img = null;
-	preg_match( '/<img[^>]+>/', $body_content, $img );
+	// Only process if there's an IMG element after this FIGURE. It should be inside it.
+	if ( ! $processor->next_tag( 'IMG' ) ) {
+		return $block_content;
+	}
```

Comment:
this will still fall on an IMG, but I'm noting that assumes that the image appears later in the document than the FIGURE, and is also assuming that an image is inside the figure.

in the HTML Processor we'd express this in the following way:
```php
if ( ! $processor->next_tag( array( 'breadcrumbs' => array( 'FIGURE', 'IMG' ) ) ) ) {
	return $block_content;
}
```

I can update the comment to make this clearer.

Reply:
expanded the comment in 0c0f5bce940e1bc516287a7fc58f0ef0d563630a